src/main/java/oof/command/StopTrackerCommand.java

```diff
@@ -26,52 +25,48 @@
     public StopTrackerCommand(String description) {
         super();
         this.description = description;
-        this.stc = new StartTrackerCommand(description);
     }
 
     /**
-     * Checks if a Start Time for Task object exists.
+     * Stops Tracker timer.
      *
-     * @param task Task object.
-     * @return boolean True if Start Time for Task object is not null.
+     * @param semesterList Instance of SemesterList that stores Semester objects.
+     * @param tasks        Instance of TaskList that stores Task objects.
+     * @param ui           Instance of Ui that is responsible for visual feedback.
+     * @param storage      Instance of Storage that enables the reading and writing of Task
+     *                     objects to hard disk.
+     * @throws OofException if invalid Module Code detected.
      */
-    private boolean isStarted(Task task) {
-        return task.getStartDate() != null;
-    }
-
-    /**
-     * Gets the time difference between the start and end of a tracked task.
-     * @param task Instance of Task object being tracked.
-     * @return number of minutes between start and end of a task.
-     * @throws OofException if start and end date is invalid.
-     */
-    private long getDateDiff(Task task) throws OofException {
-        try {
-            Date start = convertStringToDate(task.getStartDate());
-            Date end = convertStringToDate(task.getEndDate());
-            long diffInMilliSeconds = end.getTime() - start.getTime();
-            return TimeUnit.MINUTES.convert(diffInMilliSeconds, TimeUnit.MILLISECONDS);
-        } catch (ParseException e) {
-            throw new OofException("Unable to retrieve time taken. Please retry!");
-        }
-    }
-
     @Override
     public void execute(SemesterList semesterList, TaskList tasks, Ui ui, Storage storage) throws OofException {
         if (description.isEmpty()) {
-            throw new OofException("Please enter a Task!");
+            throw new OofException("Please enter an Assignment description!");
         }
-        Task task = stc.findTask(tasks, description);
-        if (task.getStatus()) {
-            throw new OofException("Task has already been completed.");
-        } else if (!isStarted(task)) {
-            throw new OofException("Task has no Start time.");
+
+        TrackerList trackerList = storage.readTrackerList();
+        Tracker tracker = trackerList.findTrackerByDesc(description);
+        boolean isCompleted = tracker.getStatus();
+
+        if (isCompleted) {
+            throw new OofException("Assignment has already been completed.");
```

Comment:
This condition is never true because of your Storage bugs.

Reply:
The code is now refactored.